src/components/sidebar/__test__/SideBar.test.tsx

```diff
@@ -0,0 +1,212 @@
+import { act, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import SideBar from '../SideBar';
+import { BrowserRouter, MemoryRouter, Route, Routes } from 'react-router-dom';
+import { AppContext } from '../../../stores/SearchContext';
+import ResultCatalog from '../../result-catalog/ResultCatalog';
+import '@testing-library/jest-dom';
+import * as apiService from '../../../service/apiService';
+// import { fetchCharacter } from '../../../service/apiService';
+
+jest.mock('../../../service/apiService');
+describe('SideBar', () => {
+  const props = {
+    startPage: false,
+    handleStopSearch: () => {},
+  };
+  const apiData = {
+    results: [
+      {
+        id: 1,
+        name: 'Morty Smith',
+        gender: 'Male',
+        species: 'Human',
+        status: 'Alive',
+        image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
+      },
+    ],
+    total: 1,
+  };
+  const context = {
+    localStorageValue: '',
+    apiData: apiData,
+    setFetchData: () => {},
+    setLocalStorageValue: () => {},
+  };
+  afterEach(jest.clearAllMocks);
+
+  test('clicking triggers an additional API call to fetch detailed information', async () => {
+    const fetchCharacterSpy = jest
+      .spyOn(apiService, 'fetchCharacter')
+      .mockResolvedValue(apiData.results[0]);
+
+    await act(async () => {
+      render(
+        <BrowserRouter>
+          <AppContext.Provider value={context}>
+            <ResultCatalog {...props}>
+              <MemoryRouter initialEntries={['/details/1']}>
+                <Routes>
+                  <Route
+                    path="details/:id"
+                    element={
+                      <SideBar
```

Comment:
The sense of unit testing is that we test a component as a unit separately from its environment. Everything from outside what impact the component should be mocked. ResultCatalog for sure is redundant here.
As I see SideBar doesn't use the context, so it's also redundant here.

Reply:
will be fixed in the next PR